src/olympia/accounts/views.py

```diff
@@ -289,11 +289,15 @@ def post(self, request):
                             status=422)
 
         user_token = os.urandom(4).encode('hex')
-        username = 'super-created-{}'.format(user_token)
+        username = serializer.data['username']
+        if not username:
```

Comment:
This is starting to feel a little tedious. You could set a `default` callable on the field but then the username and email wouldn't match up. Not sure if that matters.

It also isn't really bad and this likely isn't going to be touched again so it's fine as is.